docs/visual-basic/misc/bc30738.md

```diff
@@ -11,14 +11,44 @@ ms.assetid: c3212d87-6af3-4120-9e06-4d85fa910de2
 ---
 # 'Sub Main' is declared more than once in '\<namespace>': \<message>
 
-`Sub Main` can only be declared once inside a namespace.  
-  
- **Error ID:** BC30738  
-  
-## To correct this error  
-  
-- Make sure there is only a single `Sub Main` procedure in your project.  
-  
+`Sub Main` can only be declared once inside a namespace.
+
+**Error ID:** BC30738
+
+## Cause
+
+This error commonly occurs in Windows Forms applications when you attempt to define your own `Sub Main` procedure while the Visual Basic Application Framework is enabled. The application framework automatically generates a `Main` procedure, creating a conflict when you add another one.
+
+## To correct this error
+
+Choose one of the following approaches:
+
+### Option 1: Use the Application Framework (Recommended for most Windows Forms apps)
+
+1. Remove your custom `Sub Main` procedure.
+1. Configure your startup form and initialization code using the application framework:
+   - Set the **Startup form** in **Project Properties** > **Application** tab.
+   - Use the `My.MyApplication` events for custom startup logic.
+
+### Option 2: Disable the Application Framework
+
+If you need programmatic control over application startup (for example, to select which form to display based on command-line arguments):
+
+1. In **Project Properties** > **Application** tab, uncheck **Enable application framework**.
+1. Set **Startup object** to your module or class containing `Sub Main`.
+1. Implement your own `Sub Main` procedure to control application startup.
+
+### Option 3: Remove duplicate Main procedures
+
+- Ensure there is only a single `Sub Main` procedure in your entire project.
+
+## Accessing My.MyApplication.Main
+
+If you need to access the automatically generated startup code, you can work with the `My.MyApplication` object and its events such as `Startup`, `StartupNextInstance`, and `Shutdown`. The application framework provides these events specifically for customizing application behavior without defining your own `Main` procedure.
```

Comment:
Use `xref` syntax for `My.MyApplication`, `Startup`, `StartupNextInstance` and `Shutdown1.